react_juce/core/ReactApplicationRoot.h

```diff
@@ -63,6 +63,29 @@ namespace reactjuce
         bool keyPressed(const juce::KeyPress& key) override;
 #endif
 
+        juce::Component* componentUnderMouse;
+
+        void mouseDrag(const juce::MouseEvent &e) override {
```

Comment:
Used to translate mouseDrags into enter and exit events

Reply:
Hmm interesting, won't ReactAppRoot not get this callback if the mouse is dragged over a child component? https://docs.juce.com/master/classComponent.html#aff01e4d339bda474a66984b709044b63

Or is AppRoot getting this callback at all times?

Reply:
see View::mouseDrag I forward child drag events onto ReactAppRoot::mouseDrag.

but yes, its pretty ugly. Another way to handle it would be making ReactAppRoot a mouse listener for all its children I suppose.

Reply:
Got it, thanks! Finally coming back around to this, sorry for the delay.

So, reading this through again with a better understanding, I actually think this is quite like the way juce implements its own drag and drop patterns, except here we're sort of re-implementing it with ReactAppRoot == DragAndDropContainer and View == DragAndDropTarget. (See https://github.com/juce-framework/JUCE/blob/b8206e3604ebaca64779bf19f1613c373b9adf4f/modules/juce_gui_basics/mouse/juce_DragAndDropContainer.cpp)

This has me thinking then, especially in the context of the approach we laid out in the drag and drop issue (see https://github.com/nick-thompson/react-juce/issues/60#issuecomment-755713591), that perhaps we should just merge this goal into that work. The approach laid out in that issue is basically 1 step away from solving this problem here, and we can nudge the dragstart/dragmove/dragend callbacks that each View will receive (because its a DragAndDropTarget) to correctly send mouseenter/mouseleave callbacks as well.

What do you think?